### ISSUE TITLE
4 hours for installation

### ISSUE THREAD

**Issue by stoiximan** [foxlet/macOS-Simple-KVM#187]
Followed steps correctly but 4 hours for installation?How to speed it up?

**Comment by Synchronicity1111** [comment]
Hi @stoiximan 
What's your hardware configuration?

I ran this on a 2015 core i7 months ago and it took like an Hour & 1/2. Currently I'm testing ways to speed it up, including raid0 SSDs. Someday in the future we'd like to explore developing Paravirt drivers.

The EFI firmware **'Clover'may be mostly responsible**, As it handles all kinds of Errors during the install & bootup. You will see it printing logs & KExt (kernel module) problems to Mac's dmesg. Everytime it detects something to seems to take time? 😕

As it Clover was designed to run on real baremetal, it's defaults may not be optimal.
**It may be possible to reconfigure Clover for a simpler baseline?**

P.S. if you fork this I'd love to contribute!

**Comment by sayem314** [comment]
I guess it depends on your network speed since it downloads around 8GB while installing. For me, it took 25 minutes in total from step 1 to boot into OS on a 10GbE connection.

**Comment by stoiximan** [comment]
Ok i think it is my internet speed didnt know that it was downloading at the same time it was installing.Thanks for the quick replies

**Comment by Synchronicity1111** [comment]
Ok. That does explain a lot!
I'm on a 500Mbps business net.

We could explore Squid caching?
en.wikipedia.org/wiki/Squid_(software)

**Comment by sayem314** [comment]
The download data comes from apple servers, you can't use caching here. Also download probably happen in smaller chunks.

**Comment by xradiation** [comment]
what about the offline macos installer, i guess it would be a huge time saver; btw any idea whether that will help or not ?

**Comment by sayem314** [comment]
@xradiation Offline installer can be made, in fact I have tested this but we need to store it on the server and create another script to handle automation of this process. The offline download is 8.4GB including BaseSystem, If @foxlet is not willing to do it, I can do it if you guys can donate hosting costs :)

**Comment by xradiation** [comment]
hi @sayem314 ;
i just tested an offline installation and i can confirm that it's way faster and more stable than the online apple servers one, or at least in my case. sounds good for me however i guess the user base isn't large enough to cover all the costs as of now, so it depends on that.

**Comment by sayem314** [comment]
Yes @xradiation, offline installation is a really fast and smooth experience as well. and about offline KVM-image hosting, I will try to squeeze one of my servers and try to host if I can and in the meantime let's see what @foxlet has to say about this.

**Comment by xradiation** [comment]
alright @sayem314 in case you managed to get it done, keep us in touch man.

**Comment by sayem314** [comment]
sure thing @xradiation

**Comment by lucaszanella** [comment]
Hi @sayem314, I think this method of downloading from apple servers should is the ideal method because we don't have to trust anyone. Hope @foxlet continue to maintain the 'download from apple servers' method :)

**Comment by pikootkoh** [comment]
> hi @sayem314 ;
> i just tested an offline installation and i can confirm that it's way faster (edit : it took 15 minutes) and more stable than the online apple servers one, or at least in my case. sounds good for me however i guess the user base isn't large enough to cover all the costs as of now, so it depends on that.

Hi, may I know how to install offline. I have slow internet here and i prefer to download the files beforehand for reliable installation. 

I’m trying to install High Sierra and it keeps saying “The recovery server could not be contacted”

**Comment by FcoMarcosMAbreu** [comment]
After startup the instalation for some minutes, the entire vm stops and I'd can't finish the instalation.

**Comment by sdq99** [comment]
Hi, @xradiation 
> i just tested an offline installation and i can confirm that it's way faster (edit : it took 15 minutes) and more stable than the online apple servers one, or at least in my case. sounds good for me however i guess the user base isn't large enough to cover all the costs as of now, so it depends on that.

Can you please share any reference or a quick guid to do this. bcoz installing from internet to download around 8GB file is hard for me. and if in any case installation goes wrong, I will have to do same again and again. So it will be very helpfull for me and all those who have not enough internet connection.
or any source to download iso/dmg/img file once and boot it up when we need any fresh installation like we do for Linux?

**Comment by ghost** [comment]
> @xradiation Offline installer can be made, in fact I have tested this but we need to store it on the server and create another script to handle automation of this process. The offline download is 8.4GB including BaseSystem, If @foxlet is not willing to do it, I can do it if you guys can donate hosting costs :)



> hi @sayem314 ; i just tested an offline installation and i can confirm that it's way faster (edit : it took 15 minutes) and more stable than the online apple servers one, or at least in my case. sounds good for me however i guess the user base isn't large enough to cover all the costs as of now, so it depends on that.

Im facing this issue too. Could you please spill the tea and tell us how to achieve this?

**Comment by AlexLearnsToCode** [comment]
Any progress everybody?

**Comment by notAperson535** [comment]
@AlexLearnsToCode @FcoMarcosMAbreu @ilivss @sayem314 @stoiximan @sdq99 You can download or make a macOS ISO file from online, and just rename it to BaseSystem.img like this
```
mv ISO_NAME.iso BaseSystem.dmg
```
I just gave the same answer to someone else here #542

**Comment by sudo-nick** [comment]
@notAperson535  where did you download the macOS ISO file from? It'd be helpful if you shared any sources.

**Comment by notAperson535** [comment]
> @notAperson535 where did you download the macOS ISO file from? It'd be helpful if you shared any sources.

@sudo-nick16 I usually just look online. You could also search up "How to install macOS in VirtualBox" and find download links there

**Comment by mateusdata** [comment]
Consegui resolver o problema baixando a ISO do macOS Sequoia diretamente de:

https://archive.org/details/macos-sequoia-dev1

Após o download, renomeei o arquivo ISO para BaseSystem.img, e o OSX-KVM reconheceu e iniciou normalmente, sem necessidade de conversões adicionais.
